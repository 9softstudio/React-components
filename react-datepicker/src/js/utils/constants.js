export const MomentFormat = {
    default: "MM/DD/YYYY",
    monthSelect: "MMM"
}