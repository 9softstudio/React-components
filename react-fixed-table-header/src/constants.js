export const SCROLLBAR_WIDTH = 19;
export const BODY_WIDTH = document.body.clientWidth;
export const GUTTER = 15;
export const MAX_WIDTH = BODY_WIDTH - (GUTTER * 2);
export const DEFAULT_MILLISECOND_FOR_WAITING = 500;
export const DEFAULT_COLUMN_WIDTH = 100;