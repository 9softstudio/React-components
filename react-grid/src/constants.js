export const SCROLLBAR_WIDTH = 19;
export const GUTTER = 15;
export const DEFAULT_MILLISECOND_FOR_WAITING = 500;
export const DEFAULT_COLUMN_WIDTH = 100;