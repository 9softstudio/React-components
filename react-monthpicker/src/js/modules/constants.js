export const defaultMinYear = 2000;

export const getCurrentDate = () => new Date();