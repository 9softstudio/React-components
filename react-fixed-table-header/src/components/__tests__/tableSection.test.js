import React from 'react'
import renderer from 'react-test-renderer'

import { SCROLLBAR_WIDTH, MAX_WIDTH } from '../../constants'
import createTableSection from '../tableSection';

const defaultProps = {
    tableClass: "table",
    width: MAX_WIDTH,
    maxWidth: MAX_WIDTH,
    minWidth: undefined,
    autoWidth: true,
    maxHeight: undefined
}

describe('tableSection ', () => {
    afterAll(() => {
        jest.resetAllMocks();
    })

    describe('containerWidth getter', () => {
        it('autoWidth = true - get maxWidth value', () => {
            const expectedResult = defaultProps.maxWidth;
            const TableSection = createTableSection();

            const instance = new TableSection(defaultProps);

            const result = instance.containerWidth;

            expect(result).toBe(expectedResult);
        })

        it('autoWidth = false - get width value', () => {
            const width = 600,
                autoWidth = false;
            const props = Object.assign({}, defaultProps, { width, autoWidth })
            const expectedResult = width;
            const TableSection = createTableSection();

            const instance = new TableSection(props);
            const result = instance.containerWidth;

            expect(result).toBe(expectedResult);
        })
    })

    describe('containerStyle getter', () => {
        it('maxHeight props has value - contain maxHeight property', () => {
            const maxHeight = 300;
            const props = Object.assign({}, defaultProps, { maxHeight });
            const expectedResult = {
                width: props.maxWidth,
                minWidth: props.minWidth,
                maxHeight
            }

            const TableSection = createTableSection();
            const instance = new TableSection(props);

            const result = instance.containerStyle;

            expect(result).toEqual(expectedResult);
        })

        it('maxHeight props is undefined - do not contain maxHeight property', () => {
            const expectedResult = {
                width: defaultProps.maxWidth,
                minWidth: defaultProps.minWidth
            }

            const TableSection = createTableSection();
            const instance = new TableSection(defaultProps);

            const result = instance.containerStyle;

            expect(result).toEqual(expectedResult);
        })
    })

    describe('tableStyle getter', () => {
        it('minWidth props is undefined - style object should be correct with width = containerWidth - scrollbarWidth', () => {
            const TableSection = createTableSection();
            const instance = new TableSection(defaultProps);

            const expectedResult = { width: instance.containerWidth - SCROLLBAR_WIDTH };

            const result = instance.tableStyle;

            expect(result).toEqual(expectedResult);
        })

        it('minWidth props - style object should be correct with width = containerWidth - scrollbarWidth', () => {
            const TableSection = createTableSection();
            const instance = new TableSection(defaultProps);

            const expectedResult = { width: instance.containerWidth - SCROLLBAR_WIDTH };

            const result = instance.tableStyle;

            expect(result).toEqual(expectedResult);
        })
    })

    describe('render', () => {
        it('isHeader = true - content should render <thead> tag match with snapshot', () => {
            const extendedContainerProps = { isHeader: true };
            const TableSection = createTableSection(extendedContainerProps);
            const component = renderer.create(<TableSection {...defaultProps}><th>Col1</th><th>Col2</th></TableSection>);

            const result = component.toJSON();

            expect(result).toMatchSnapshot();
        })

        it('isHeader = false - content should render <tbody> tag match with snapshot', () => {
            const extendedContainerProps = { isHeader: false };
            const TableSection = createTableSection(extendedContainerProps);
            const component = renderer.create(<TableSection {...defaultProps}><td>Data1</td><td>Data2</td></TableSection>);

            const result = component.toJSON();

            expect(result).toMatchSnapshot();
        })
    })
})