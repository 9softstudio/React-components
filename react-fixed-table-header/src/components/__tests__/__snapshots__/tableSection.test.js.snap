// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tableSection  render isHeader = false - content should render <tbody> tag match with snapshot 1`] = `
<div
  style={
    Object {
      "maxHeight": undefined,
      "minWidth": undefined,
      "width": -30,
    }
  }
>
  <div
    className="table-wrapper"
  >
    <table
      className="table"
      style={
        Object {
          "width": -49,
        }
      }
    >
      <tbody>
        <td>
          Data1
        </td>
        <td>
          Data2
        </td>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`tableSection  render isHeader = true - content should render <thead> tag match with snapshot 1`] = `
<div
  style={
    Object {
      "maxHeight": undefined,
      "minWidth": undefined,
      "width": -30,
    }
  }
>
  <div
    className="table-wrapper"
  >
    <table
      className="table"
      style={
        Object {
          "width": -49,
        }
      }
    >
      <thead>
        <th>
          Col1
        </th>
        <th>
          Col2
        </th>
      </thead>
    </table>
  </div>
</div>
`;
